src/cameras/PerspectiveCamera.js

```diff
@@ -99,6 +108,59 @@ class PerspectiveCamera extends Camera {
 
 	}
 
+	/*
+	 * Provides the topRight and bottomLeft corners of the camera's 2D frustum slice at a given distance.
+	 * Results are copied into minTarget and maxTarget input vars.
+	 */
+	getBounds( distance, minTarget, maxTarget ) {
+
+		_tempV3.set( - 1, - 1, 0.5 ).applyMatrix4( this.projectionMatrixInverse );
+		_tempV3.multiplyScalar( distance / Math.abs( _tempV3.z ) );
+		minTarget.x = _tempV3.x;
+		minTarget.y = _tempV3.y;
+
+		_tempV3.set( 1, 1, 0.5 ).applyMatrix4( this.projectionMatrixInverse );
+		_tempV3.multiplyScalar( distance / Math.abs( _tempV3.z ) );
+		maxTarget.x = _tempV3.x;
+		maxTarget.y = _tempV3.y;
+
+	}
+
+	/*
+	 * Calculates the height/width of the camera's frustum at a given distance.
+	 * returns a Vector2 where x is width and y is height.
+	 */
+	frustumDimensions( distance ) {
+
+		this.getBounds( distance, _minTarget, _maxTarget );
+		_dimensions.x = _maxTarget.x - _minTarget.x;
+		_dimensions.y = _maxTarget.y - _minTarget.y;
+
+		return _dimensions.clone();
```

Comment:
Had intentionally chosen return the clone from a temp vector since it seemed like a convenience/speed tradeoff where the speed impact is extremely low? Feel like a method that returns the value is more convenient/intuitive than a method that mutates an input arg? 

Not opposed to the change, just want to understand the logic.

Reply:
It is the three.js policy. It is up to the app to allocate the memory.

Reply:
Rodger that, sounds good!

Reply:
- target `Vector2` added here https://github.com/mrdoob/three.js/pull/27574/commits/27a49c62681596de68fcec1a620e09b7d5802f0b
- Fixed name + updated comment here  https://github.com/mrdoob/three.js/pull/27574/commits/fcab13fae3291158c6729cbd52d310bd623acd61